src/modules/user/user.service.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  Inject,
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { UserRepository } from './user.repository';
+import { User } from '@prisma/client';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @Inject('UserRepository') private readonly userRepository: UserRepository,
+  ) {}
+
+  async login(email: string, password: string): Promise<User | null> {
+    const user = await this.userRepository.findByEmail(email);
+
+    if (!user) {
```

Comment:
Prisma client should expose a function called findOneOrThrow or something along those lines, please use that one, no need to reinvent the wheel.